src/pages/home/report/EmojiPickerMenu.js

```diff
@@ -0,0 +1,122 @@
+import React, {Component} from 'react';
+import {View, FlatList, Text} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+import emojis from '../../../../assets/emojis';
+import EmojiPickerMenuItem from './EmojiPickerMenuItem';
+import TextInputFocusable from '../../../components/TextInputFocusable';
+
+const propTypes = {
+    // Controls the visibility of this component.
+    isVisible: PropTypes.bool,
+
+    // Function to add the selected emoji to the main compose text input
+    addEmojiToTextBox: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    isVisible: false,
+};
+
+class EmojiPickerMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        // Ref for the emoji search input
+        this.searchInput = undefined;
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 500, false);
+        this.renderItem = this.renderItem.bind(this);
+
+        this.state = {
+            filteredEmojis: emojis,
+
+            // This is the indices of each category of emojis
+            // The positions are static, and are calculated as index/numColumns (8 in our case)
+            // This is because each row of 8 emojis counts as one index
+            headerIndices: [0, 34, 60, 88, 99, 121, 148],
+        };
+    }
+
+    componentDidUpdate() {
+        if (this.props.isVisible) {
+            this.searchInput.focus();
+        }
+    }
```

Comment:
The idea is that each time the picker opens we want to focus it

Reply:
If we want to focus as a side effect of gaining or losing visibility then it would be more correct to either:

- Focus when the component mounts and mount it each time it opens
- Track the visibility and update it by looking to see if it was previously not visible and now visible

Reply:
I added a prop for isVisible and changed it to `componentDidMount`, it hits that line but isn't focusing for some reason. Any guesses?

Reply:
Probably because it only renders once when inside a Modal so we would have to conditionally render the modal child?

Reply:
Hmm it was hitting the line every time I opened the modal though (i put a log line there to test)

Reply:
It could also be that the element isn't ready yet. I've ran into this before where if an animation was still in progress the ref wasn't available until after the animation is complete.

Reply:
Hm gotcha. So what would be the solution then? Is there a way to wait until it is rendered?

Reply:
Idk tbh I'd first verify that was happening. Really just making guesses about what the issue could be.